pkgs/by-name/fa/fasthenry/package.nix

```diff
@@ -17,53 +17,42 @@ stdenv.mkDerivation rec {
   src = fetchFromGitHub {
     owner = "ediloren";
     repo = "FastHenry2";
-    rev = "R${version}";
-    sha256 = "017kcri69zhyhii59kxj1ak0gyfn7jf0qp6p2x3nnljia8njdkcc";
+    rev = "R${finalAttrs.version}";
+    hash = "sha256-jM0mLVJRUmtHF9dcDJw81vkHpgqyz1RihB7+ZGJm8wQ=";
   };
 
-  dontConfigure = true;
+  env.NIX_CFLAGS_COMPILE = toString [
+    "-fcommon -O -DFOUR"
+    (lib.optional stdenv.hostPlatform.isx86_64 "-m64")
+    "-Wno-error=implicit-function-declaration"
+    "-Wno-error=implicit-int"
+    "-Wno-error=return-mismatch"
+  ];
 
-  preBuild =
-    ''
-      makeFlagsArray=(
-        CC="gcc"
-        RM="rm"
-        SHELL="sh"
-        "all"
-      )
-    ''
-    + (
-      if stdenv.hostPlatform.isx86_64 then
-        ''
-          makeFlagsArray+=(
-            CFLAGS="-fcommon -O -DFOUR -m64"
-          );
-        ''
-      else
-        ''
-            makeFlagsArray+=(
-              CFLAGS="-fcommon -O -DFOUR"
-          );
-        ''
-    );
+  makeFlags = [ "all" ]; # need "all" to be explicitely set
 
   installPhase = ''
+    runHook preInstall
+
     mkdir -p $out/bin
     cp -r bin/* $out/bin/
-    mkdir -p $out/share/doc/${pname}-${version}
-    cp -r doc/* $out/share/doc/${pname}-${version}
-    mkdir -p $out/share/${pname}-${version}/examples
-    cp -r examples/* $out/share/${pname}-${version}/examples
+    mkdir -p $out/share/doc/
+    cp -r doc/* $out/share/doc/
+    mkdir -p $out/share/examples
+    cp -r examples/* $out/share/examples
```

Comment:
I *think* it's preferred to have an extra layer in `share/doc/` so that there are no file collisions in case the share directories were to be merged together with other packages.
In this case, I think `share/doc/fasthenry/` would be fine.

I guess `share/examples/` has an even higher chance of collision than docs...

Though, I don't know the exact policy about this... so, do whatever you think is better.

Reply:
the extra layer in `share/doc` and `share/examples` were introduced to avoid collisions as mentioned by @TomaSajt